src/core/core.scale.js

```diff
@@ -60,6 +60,111 @@ module.exports = function(Chart) {
+	 * @return object with text placment origin coordinate and alignment
+	 */
+	function useEndOnTruncate(isForward, centerPoint, halfWidth, maxExtent) {
+		var positionProperties = {
+			coord: centerPoint,
+			alignment: 'center'
+		};
+		// Do the start trunction check first to get 'start' alignment when both
+		// ends would truncate
+		if (isForward) {// right side up text
+			if (centerPoint < halfWidth) {// would truncate the start
+				positionProperties.coord = 0;
+				positionProperties.alignment = 'start';
+			} else if (centerPoint + halfWidth > maxExtent) {// would truncate the end
+				positionProperties.coord = maxExtent;
+				positionProperties.alignment = 'end';
+			}// end of (forward) text truncation checks
+		} else if (centerPoint + halfWidth > maxExtent) {// would truncate the start
+			positionProperties.coord = maxExtent;
+			positionProperties.alignment = 'start';
+		} else if (centerPoint < halfWidth) {// would truncate the end
+			positionProperties.coord = 0;
+			positionProperties.alignment = 'end';
+		}// end of (reverse) text truncation checks
+
+		return positionProperties;
+	}// ./useEndOnTruncate(…){
+
+	/**
+	 * Place text, centered if it will fit
+	 *
+	 * Center text, if it fully fits in an area plus margins, otherwise set
+	 *	origin to the outer margin edge, and make that the start/end of the text
+	 *
+	 * Font must be set for the context before calling.	Fails to set (at least)
+	 *	the size if passed in the constraints, and set here.
+	 * Handles horizontal or vertical placment, but not other angles
+	 *
+	 * @param ctx CanvasRenderingContext2D drawing context
+	 * @param text single text string
+	 * @param cnstr object with text placement contraints
+	 * @return object with adjusted text placment properties
+	 */
+	function constrainedTextCenter(ctx, text, cnstr) {
+		var txtProperties;
+		var isForward = cnstr.position !== 'left'; // All text is right side up,
+			// except for Y Axis placed on the left of the chart
+		var halfTextWidth = ctx.measureText(text).width / 2;
+		var out = {
+			positionX: 0,
+			positionY: 0,
+			rotation: 0,
+			hAlign: 'center'
+		};
+
+		if (cnstr.isHorizontal) {// is horizontal text
+			txtProperties = useEndOnTruncate(isForward,
+				midPoint(cnstr.container.left, cnstr.container.right),
+				halfTextWidth,
+				cnstr.container.right + cnstr.container.margins.right);
+			out.positionX = txtProperties.coord;
+
+			out.positionY = cnstr.position === 'bottom' ?
+				cnstr.container.bottom - cnstr.verticalOffset :
+				cnstr.container.top + cnstr.verticalOffset;
+		} else {// not (cnstr.isHorizontal) // is vertical text
+			txtProperties = useEndOnTruncate(isForward,
+				midPoint(cnstr.container.top, cnstr.container.bottom),
+				halfTextWidth,
+				cnstr.container.bottom + cnstr.container.margins.bottom);
+			out.positionY = txtProperties.coord;
+
+			out.positionX = isForward ? // is text on the right side of the chart ?
+				cnstr.container.right - cnstr.verticalOffset :
```

Comment:
`baseLineOffset` is fine for me (would personally pick `lineOffset`, not sure what the value of `base`). And yes, when object properties, excessive descriptive names are a problem since they can't be minified.